app/controllers/admin/application_controller.rb

```diff
@@ -10,21 +10,23 @@ class ApplicationController < Administrate::ApplicationController
     before_action :authorized_or_redirect
     before_action :set_paper_trail_whodunnit
 
+    private
+
     def require_user
       return if current_user
 
       redirect_to login_path, alert: 'Please sign in to continue'
     end
 
     def authorized_or_redirect
-      return if can?(action_name, resource_name)
+      return if authorized_action?(resource_name, action_name)
 
       redirect_to root_path, alert: 'Not authorized.'
     end
 
     # Hide links to actions if the user is not allowed to do them.
     # This is an override of an Administrate method to work with CanCan
-    def show_action?(action, resource)
+    def authorized_action?(resource, action)
```

Comment:
Non-blocking: I see here, and in the underlying PR against the gem, that some orders of arguments are changing. This is a bit confusing to me, as the `can?` call is still in the original order, but now some methods go the other way.

I don't think there's anything to be done about this here - I see that `authorized_action?` here, and `accessible_action?` in the views are now consistent in their argument use, which probably hides the complexity well enough - and I don't want to bike-shed this. I'm just sure that this will trip someone up in the future, and I don't look forward to that being me.

Reply:
Yeah, I hear you. It seemed less confusing to me to match the argument order with what's in the upstream code, but you're absolutely right that this is likely to confuse future us anyway. Fwiw, I tend to think that will unfortunately be the case with any administrate update that involves changing custom controller methods and views.